docs/api.rst

```diff
@@ -5,11 +5,43 @@ Hyperlink API
 
 .. automodule:: hyperlink._url
 
+.. contents::
+   :local:
+
 Creation
 --------
 
-Before you can work with URLs, you must create URLs. There are two
-ways to create URLs, from parts and from text.
+Before you can work with URLs, you must create URLs.
+
+Parsing Text
+^^^^^^^^^^^^
+
+If you already have a textual URL, the easiest way to get URL objects
+is with the :func:`parse()` function:
+
+.. autofunction:: hyperlink.parse
+
+By default, :func:`~hyperlink.parse()` returns an instance of
+:class:`DecodedURL`, a URL type that handles all encoding for you, by
+wrapping the lower-level :class:`URL`.
+
+DecodedURL
+^^^^^^^^^^
+
+.. autoclass:: hyperlink.DecodedURL
+.. automethod:: hyperlink.DecodedURL.from_text
+
+The Encoded URL
+^^^^^^^^^^^^^^^
+
+The lower-level :class:`URL` looks very similar to the
+:class:`DecodedURL`, but does not handle all encoding cases for
+you. Use with caution.
+
+.. note::
+
+   :class:`URL` is also available as an alias,
+   ``hyperlink.EncodedURL`` for more explicit usage.
 
 .. autoclass:: hyperlink.URL
 .. automethod:: hyperlink.URL.from_text
```

Comment:
What do people think about switching over all "automethods" below this point to be DecodedURL methods?

For my money, I think it makes sense to have only one set of the URL methods given that the names are all the same, and there's only one behaviorial difference that they all share. Since DecodedURL is the recommended API, shouldn't all the methods point at that? Maybe for another PR though.

Reply:
I think it would be helpful to define an ABC which has the shared API and the docs can go there.

Doing so would help us catch any inconsistencies in that shared API between the two classes, and it would allow one to specify a type hint for an object that can be either, etc.

Reply:
I agree an ABC could be useful for some. We'd have to call it AbstractUR, maybe BaseURL, or something, because I don't want to have to explain yet another URL type in the docs.

If everyone's ok with that direction, I would propose the ABC for a separate PR, and we can update the docs the rest of the way then.

Reply:
Yup to `AbstractURL`, since `URL` is taken.

Reply:
Resolving by way of opening #127 :)